test/clj/cider/nrepl/middleware/inspect_test.clj

```diff
@@ -505,3 +518,17 @@
                                               :var-name "sub-map"})
                             (session/message {:op   "eval"
                                               :code "sub-map"}))))))))
+
+(deftest inspect-tap-current-value-test
+  (testing "inspect-tap-current-value taps the current inspector value"
+    (do
+      (session/message {:op   "eval"
+                        :code "(def x (+ 3 4)))"})
+      (session/message {:op "eval"
+                        :inspect "true"
+                        :code    "x"})
+      (session/message {:op  "inspect-push"
+                        :idx 1})
+      (session/message {:op  "inspect-tap-current-value"}))
```

Comment:
I'd recommend to wait untiil `(= "7"  @inspect-tap-current-value-test-atom)`, up to 10 seconds, then perform the `is`.

What is probably happening is that the clojure taps aren't instantaneous since they have some threading stuff behind the scenes.

(In Orchard we also have some test flakiness related to the inspector/tap functionality. I never got to debug it)

...There's the possibility that we'll still have flakiness after the improvement, but not as pronounced as right now.

Reply:
that fixed it